### PR TITLE
Fix accessibility errors and low-contrast text on file/image upload fields

### PR DESCRIPTION
**Pull request by stefan-cotitosu**
## Summary

Fixes the WAVE accessibility findings from Codeinwp/ppom-pro#284 (missing label on the file input, low-contrast text), plus a few more of the same class found while reproducing it: an orphaned label on the Image field, empty labels on image swatch options, and low-contrast description text.

- **File / Image field captions**: `<label for="{data_name}">` pointed at an id no control in the field actually carries (each image option gets its own id; the file field's real control is injected by Plupload at runtime). Switched to `<fieldset>`/`<legend>`, the correct native pattern for captioning a grouped/composite control. Added a CSS reset so layout is unchanged.
- **Image swatch options**: `label.ppom-palette-item` wrapped only a checkbox + ``, no text content ("Empty form label"). Added a visually-hidden `.screen-reader-text` span with the option's title.
- **Native file input**: Plupload's HTML5 runtime injects a real, invisible `input[type=file]` over the "Select files" button with no accessible name ("Missing form label" — the original report in #284). Set `aria-label` on it in `PostInit`, once Plupload has created it, using the field title.
- **Description text contrast**: `.show_description` was `#999` on white (2.84:1, fails WCAG AA). Darkened to `#666` (~5.7:1).
- **Upload button contrast**: `.btn-primary` reuses a generic Bootstrap class name that some themes (e.g. Hestia Pro) also define, with a lighter accent color that wins the cascade over this plugin's bundled `bootstrap.css`. Forced a compliant blue so contrast holds regardless of the active theme.

## Test plan

- [x] Reproduced all findings with the WAVE browser extension on a product with a conditional Image + File Upload field, under both Hestia Pro and Neve.
- [x] Re-ran WAVE after each fix; each targeted error cleared.
- [x] `php -l` on both changed templates; `node --check` on the changed JS.
- [ ] Manual visual check across more themes than Hestia Pro / Neve (not done here).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: phpcs.xml [hunk 1]
```diff
@@ -61,6 +61,7 @@
         <properties>
             <property name="customAutoEscapedFunctions" type="array">
                 <element value="neve_kses_svg" />
+                <element value="ppom_esc_html" />
```
[the remark this answers is not in the document]
Comment (stefan-cotitosu) [reply phpcs.xml]: Good catch, reverted. Validator::esc_html() does mutate $allowedposttags to allow <script>/<iframe>/onclick etc before wp_kses() -- registering it globally would've silenced this sniff for it everywhere, including the ~172 other findings already in phpcs.baseline.xml. Scoped a phpcs:ignore to just the two lines this PR touches instead, with a comment explaining why (matches the existing, if imperfect, project pattern for this value elsewhere). https://github.com/Codeinwp/woocommerce-product-addon/pull/740/commits/7d86846

File: js/file-upload.js [hunk 2]
```diff
@@ -663,6 +663,31 @@ function ppom_setup_file_upload_input( file_input ) {
...
+				// (invisible, stacked over the "Select files" button) into our
+				// `container` element with no accessible name of its own.
+				// Give it one now that it exists. Read the name from what's
+				// actually rendered (the field legend, then the chooser
+				// button) rather than `file_input.title`: that's the raw
+				// stored metadata, not the translated/filtered/escaped text
+				// a sighted shopper sees.
+				const nativeFileInput = document.querySelector(
+					`#ppom-file-container-${ file_data_name } input[type="file"]`
+				);
```
[the remark this answers is not in the document]
Comment (stefan-cotitosu) [reply js/file-upload.js]: Good catch. Fixed: querySelectorAll + take the last match instead of the first -- each PostInit only ever runs right after its own shim has just been appended, so the last match in the container is always the instance currently initializing, whether it's the original uploader or cropper's 'Change image' one. Also flipped the accessible-name fallback to prefer the chooser button's own text (e.g. 'Change image') over the shared field legend, since it's more specific per instance.

Didn't add the two-input cropper regression test -- that needs a cropper field builder fixture (none exists in tests/e2e/fixtures yet) plus driving the upload -> crop -> 'Change image' flow, and I have no way to run Playwright against a live environment from here to verify it actually passes. Happy to add it if someone can iterate on it live, or if there's a way to run this suite I'm missing. https://github.com/Codeinwp/woocommerce-product-addon/pull/740/commits/7d86846

File: tests/e2e/specs/file-upload.spec.js [hunk 3]
```diff
@@ -1190,3 +1190,54 @@ test.describe( 'File Upload with Dynamic Nonce Refresh', () => {
...
 } );
+
+test.describe( 'File Upload Accessibility', () => {
+	/**
+	 * Regression test for Codeinwp/ppom-pro#284: WAVE flagged the native
+	 * input Plupload injects over the "Select files" button ("Missing form
+	 * label"), and the field's own caption as an orphaned label[for] pointing
+	 * at an id no control in the field carries ("Orphaned form label").
+	 */
+	test( 'file field caption is a fieldset/legend and the native file input has an accessible name', async ( {
```
[the remark this answers is not in the document]
Comment (stefan-cotitosu) [reply tests/e2e/specs/file-upload.spec.js:1201]: Fair -- same gap I flagged myself on the earlier version of this PR. Traced through Support/Helpers::convert_options_to_key_val() and Hooks/Callbacks::update_converted_option_keys() to work out a minimal images[] shape that wouldn't need a real WP media attachment (title/id/url), but I can't actually run this test suite from here (no live environment to iterate against), so I'm not going to commit an unverified spec that might just be a second failing check. Same offer as above: happy to add coverage for both the modern and legacy_view Image variants if someone can run it live.

**Review by Soare-Robert-Daniel — APPROVED** [review]
